c7n/resources/iam.py

```diff
@@ -1108,83 +1108,118 @@ class UserDelete(BaseAction):
-        for r in resources:
-            self.log.debug('Deleting user %s' % r['UserName'])
-            self.process_user(client, r)
 
-    def process_user(self, client, r):
+    def delete_console_access(self, client, r):
         try:
             client.delete_login_profile(
                 UserName=r['UserName'])
         except ClientError as e:
             if e.response['Error']['Code'] not in ('NoSuchEntity',):
                 raise
 
-        # Access Keys
+    def delete_access_keys(self, client, r):
         response = client.list_access_keys(UserName=r['UserName'])
         for access_key in response['AccessKeyMetadata']:
             client.delete_access_key(UserName=r['UserName'],
                                      AccessKeyId=access_key['AccessKeyId'])
 
-        # User Policies
+    def delete_user_policies(self, client, r):
         response = client.list_attached_user_policies(UserName=r['UserName'])
         for user_policy in response['AttachedPolicies']:
             client.detach_user_policy(
                 UserName=r['UserName'], PolicyArn=user_policy['PolicyArn'])
 
-        # HW MFA Devices
+    def delete_hw_mfa_devices(self, client, r):
         response = client.list_mfa_devices(UserName=r['UserName'])
         for mfa_device in response['MFADevices']:
             client.deactivate_mfa_device(
                 UserName=r['UserName'], SerialNumber=mfa_device['SerialNumber'])
 
-        # Groups
+    def delete_groups(self, client, r):
         response = client.list_groups_for_user(UserName=r['UserName'])
         for user_group in response['Groups']:
             client.remove_user_from_group(
                 UserName=r['UserName'], GroupName=user_group['GroupName'])
 
-        # SSH Keys
+    def delete_ssh_keys(self, client, r):
         response = client.list_ssh_public_keys(UserName=r['UserName'])
         for key in response.get('SSHPublicKeys', ()):
             client.delete_ssh_public_key(
                 UserName=r['UserName'], SSHPublicKeyId=key['SSHPublicKeyId'])
 
-        # Signing Certificates
+    def delete_signing_certificates(self, client, r):
         response = client.list_signing_certificates(UserName=r['UserName'])
         for cert in response.get('Certificates', ()):
             client.delete_signing_certificate(
                 UserName=r['UserName'], CertificateId=cert['CertificateId'])
 
+    def delete_service_specific_credentials(self, client, r):
         # Service specific user credentials (codecommit)
         response = client.list_service_specific_credentials(UserName=r['UserName'])
         for screds in response.get('ServiceSpecificCredentials', ()):
             client.delete_service_specific_credential(
                 UserName=r['UserName'],
                 ServiceSpecificCredentialId=screds['ServiceSpecificCredentialId'])
 
+    def delete_user(self, client, r):
         client.delete_user(UserName=r['UserName'])
 
+    OPTIONS = {
```

Comment:
can you move the class vars above the methods in the class def

Reply:
I originally had them above but the reference to `OPTIONS.keys` in schema required it to be below the method definitions. I'm thinking of moving the `delete_*` outside of the class itself, as `delete_access_keys` could actually be reused in the `remove-keys`

Reply:
ended up switching to use getattr and using a dict of `{str:str}` to be able to move the class vars to the top while keeping the methods inside of the class for encapsulation